### PR TITLE
Transactions

### PR DESCRIPTION
**Pull request by andreaspsson**
Now possible to create transactions that can read, commit and delete multiple entities within one single transaction.

Have a few ideas on how to change the transaction API, som maybe there will be some changes in a near future

The transaction commit method does not return the entities right now, this is because:
1. I did not find a good way to do it
2. It would have required the TransactionalConnection to clone ll entities in the transaction in order to have some data to return
3. I think the most common case is to do stuff and then commit the transaction once done :)

### CODE REVIEW

File: src/transaction.rs [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+    let transaction_id = begin_transaction.transaction.ok_or(DatastoreClientError::ApiDataError)?;
+    Ok(TransactionConnection {
+      connection,
+      transaction_id,
+      mutations: vec![],
+    })
+  }
+
+  pub fn save(&mut self, item: impl Into<DatastoreEntity>) -> Result<(), DatastorersError> {
```
Comment (bes) [inline src/transaction.rs:63]: call this one `push_save`?

File: src/transaction.rs [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    let mut mutation = Mutation::default();
+    mutation.upsert = Some(ent);  
+    mutation.base_version = base_version;
+
+    self.mutations.push(mutation);
+
+    Ok(())
+  }
+
+  pub fn delete(&mut self, item: impl Into<DatastoreEntity>) -> Result<(), DatastorersError> {
```
Comment (bes) [inline src/transaction.rs:77]: call this one `push_delete`?

Comment (andreaspsson) [reply src/transaction.rs:77]: yes, makes sense

Comment (andreaspsson) [reply src/transaction.rs:77]: will do, same for save

**Pull request merged**